controllers/managedocs_controller_test.go

```diff
@@ -389,6 +389,25 @@ var _ = Describe("ManagedOCS controller", func() {
 				Expect(k8sClient.Delete(ctx, secret)).Should(Succeed())
 			})
 		})
+		When("there is no enable-mcg field in the add-on parameters secret", func() {
+			It("should not create a reconciled resources", func() {
+				// Create  add-on parameters secret with size field only
+				secret := addonParamsSecretTemplate.DeepCopy()
+				secret.Data["size"] = []byte("1")
+				Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+
+				// Ensure, over a period of time, that the resources are not created
+				resList := []runtime.Object{
+					scTemplate.DeepCopy(),
+					promTemplate.DeepCopy(),
+					amTemplate.DeepCopy(),
+				}
+				utils.EnsureNoResources(k8sClient, ctx, resList, timeout, interval)
+
+				// Remove the secret for future cases
+				Expect(k8sClient.Delete(ctx, secret)).Should(Succeed())
+			})
+		})
```

Comment:
After hardcoding the `enable-mcg` param to false, this test should fail.

Reply:
modified the test in the next patch